execution/binary/scalar.go

```diff
@@ -138,6 +140,19 @@ func (o *scalarOperator) Next(ctx context.Context) ([]model.StepVector, error) {
 			}
 			step.AppendSample(o.pool, vector.SampleIDs[i], val)
 		}
+
+		for i := range vector.HistogramIDs {
+			scalarVal := math.NaN()
```

Comment:
couldnt we move `scalarVal` out of the `for i := range` loop since its not dependent on i and used for samples and histograms?

Reply:
Good point, updated 👍